src/core/public/application/application_service.test.ts

```diff
@@ -924,6 +924,63 @@ describe('#start()', () => {
       await navigateToApp('baseApp:legacyApp1');
       expect(setupDeps.redirectTo).toHaveBeenCalledWith('/test/app/baseApp');
     });
+
+    describe('when `replace` option is true', () => {
+      it('use `history.replace` instead of `history.push`', async () => {
+        service.setup(setupDeps);
+
+        const { navigateToApp } = await service.start(startDeps);
+
+        await navigateToApp('myTestApp', { replace: true });
```

Comment:
Only reproduced/added a subset of the `navigateToApp` test suite, as the suite is big and as it's internally using the same logic.